src/Analyzers/CSharp/Analyzers/UseParameterNullChecking/CSharpUseParameterNullCheckingDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,253 @@
+                if (argumentNullException is null)
+                {
+                    return;
+                }
+
+                var argumentNullExceptionConstructor = argumentNullException.InstanceConstructors.FirstOrDefault(m =>
+                    m.DeclaredAccessibility == Accessibility.Public
+                    && m.Parameters.Length == 1
+                    && m.Parameters[0].Type.SpecialType == SpecialType.System_String);
+                if (argumentNullExceptionConstructor is null)
+                {
+                    return;
+                }
+
+                var objectType = compilation.GetSpecialType(SpecialType.System_Object);
+                var referenceEqualsMethod = (IMethodSymbol?)objectType
+                    .GetMembers(nameof(ReferenceEquals))
+                    .FirstOrDefault(m => m is IMethodSymbol { DeclaredAccessibility: Accessibility.Public, Parameters.Length: 2 });
+
+                context.RegisterSyntaxNodeAction(
+                    context => AnalyzeSyntax(context, argumentNullExceptionConstructor, referenceEqualsMethod),
+                    SyntaxKind.ConstructorDeclaration,
+                    SyntaxKind.MethodDeclaration,
+                    SyntaxKind.LocalFunctionStatement,
+                    SyntaxKind.SimpleLambdaExpression,
+                    SyntaxKind.ParenthesizedLambdaExpression,
+                    SyntaxKind.AnonymousMethodExpression);
+            });
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, IMethodSymbol argumentNullExceptionConstructor, IMethodSymbol? referenceEqualsMethod)
+        {
+            var cancellationToken = context.CancellationToken;
+
+            var semanticModel = context.SemanticModel;
+            var syntaxTree = semanticModel.SyntaxTree;
+
+            var option = context.Options.GetOption(CodeStyleOptions2.PreferParameterNullChecking, semanticModel.Language, syntaxTree, cancellationToken);
+            if (!option.Value)
+            {
+                return;
+            }
+
+            var node = context.Node;
+            var block = node switch
+            {
+                MethodDeclarationSyntax methodDecl => methodDecl.Body,
+                ConstructorDeclarationSyntax constructorDecl => constructorDecl.Body,
+                LocalFunctionStatementSyntax localFunctionStatement => localFunctionStatement.Body,
+                AnonymousFunctionExpressionSyntax anonymousFunction => anonymousFunction.Block,
+                _ => throw ExceptionUtilities.UnexpectedValue(node)
+            };
+            if (block is null)
+            {
+                return;
+            }
+
+            var methodSymbol = node is AnonymousFunctionExpressionSyntax
+                ? semanticModel.GetSymbolInfo(node, cancellationToken).Symbol
+                : semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            if (methodSymbol is null)
```

Comment:
No functional change intended, but just a quick return instead of unnecessarily analyzing the Block statements.

```suggestion
            if (methodSymbol is null || methodSymbol.Parameters.Length == 0)
```